rewrite-maven/src/main/java/org/openrewrite/maven/AddParentPom.java

```diff
@@ -0,0 +1,194 @@
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.marker.SearchResult;
+import org.openrewrite.maven.table.MavenMetadataFailures;
+import org.openrewrite.maven.tree.MavenMetadata;
+import org.openrewrite.maven.tree.MavenRepository;
+import org.openrewrite.semver.Semver;
+import org.openrewrite.semver.VersionComparator;
+import org.openrewrite.xml.AddToTagVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddParentPom extends Recipe {
+    transient MavenMetadataFailures metadataFailures = new MavenMetadataFailures(this);
+
+    @Option(displayName = "Group ID",
+            description = "The group ID of the maven parent pom to be adopted.",
+            example = "org.springframework.boot")
+    String groupId;
+
+    @Option(displayName = "Artifact ID",
+            description = "The artifact ID of the maven parent pom to be adopted.",
+            example = "spring-boot-starter-parent")
+    String artifactId;
+
+    @Option(displayName = "Version",
+            description = "An exact version number or node-style semver selector used to select the version number.",
+            example = "29.X")
+    String version;
+
+    @Option(displayName = "Relative path",
+            description = "New relative path attribute for parent lookup.",
+            example = "../pom.xml")
+    @Nullable
+    String relativePath;
+
+    @Option(displayName = "Version pattern",
+            description = "Allows version selection to be extended beyond the original Node Semver semantics. So for example," +
+                          "Setting 'version' to \"25-29\" can be paired with a metadata pattern of \"-jre\" to select Guava 29.0-jre",
+            example = "-jre",
+            required = false)
+    @Nullable
+    String versionPattern;
+
+    @Option(displayName = "File pattern",
+            description = "A glob expression that can be used to constrain which directories or source files should be searched. " +
+                          "Multiple patterns may be specified, separated by a semicolon `;`. " +
+                          "If multiple patterns are supplied any of the patterns matching will be interpreted as a match. " +
+                          "When not set, all source files are searched. ",
+            required = false,
+            example = "**/*-parent/grpc-*/pom.xml")
+    @Nullable
+    String filePattern;
+
+    @Override
+    public String getDisplayName() {
+        return "Change Maven parent";
+    }
+
+    @Override
+    public String getInstanceNameSuffix() {
+        return String.format("`%s:%s:%s`", groupId, artifactId, version);
+    }
+
+    @Override
+    public String getDescription() {
+        return "Change the parent pom of a Maven pom.xml. Identifies the parent pom to be changed by its groupId and artifactId.";
```

Comment:
Perhaps good to note here what our handling is in the case of an existing parent, or a multi module project. That would help clarify the usage patterns we expect to see, and the changes users can expect from the recipe.

Reply:
Right, I forgot that :)

Do we need to be explicit about multi module project? If the recipe find a pom.xml with a parent already defined, it'll ignore it. So, in a multi module project, it won't add to the child modules as they already have a parent.